src/component/contents/exploration/DraftEditorBlockNode.react.js

```diff
@@ -0,0 +1,370 @@
+
+  return nextSiblingKey
+    ? contentState.getBlockForKey(nextSiblingKey).getType() === block.getType()
+    : false;
+};
+
+const applyWrapperElementToSiblings = (
+  wrapperTemplate: *,
+  Element: string,
+  nodes: Array<React.Node>,
+): Array<React.Node> => {
+  const wrappedSiblings = [];
+
+  // we check back until we find a sibbling that does not have same wrapper
+  for (const sibling: any of nodes.reverse()) {
+    if (sibling.type !== Element) {
+      break;
+    }
+    wrappedSiblings.push(sibling);
+  }
+
+  // we now should remove from acc the wrappedSiblings and add them back under same wrap
+  nodes.splice(nodes.indexOf(wrappedSiblings[0]), wrappedSiblings.length + 1);
+
+  const childrenIs = wrappedSiblings.reverse();
+
+  const key = childrenIs[0].key;
+
+  nodes.push(
+    React.cloneElement(
+      wrapperTemplate,
+      {
+        key: `${key}-wrap`,
+        'data-offset-key': DraftOffsetKey.encode(key, 0, 0),
+      },
+      childrenIs,
+    ),
+  );
+
+  return nodes;
+};
+
+const getDraftRenderConfig = (
+  block: BlockNodeRecord,
+  blockRenderMap: DraftBlockRenderMap,
+): * => {
+  const configForType =
+    blockRenderMap.get(block.getType()) || blockRenderMap.get('unstyled');
+
+  const wrapperTemplate = configForType.wrapper;
+  const Element =
+    configForType.element || blockRenderMap.get('unstyled').element;
+
+  return {
+    Element,
+    wrapperTemplate,
+  };
+};
+
+const getCustomRenderConfig = (
+  block: BlockNodeRecord,
+  blockRendererFn: BlockRenderFn,
+): * => {
+  const customRenderer = blockRendererFn(block);
+
+  if (!customRenderer) {
+    return {};
+  }
+
+  const {
+    component: CustomComponent,
+    props: customProps,
+    editable: customEditable,
+  } = customRenderer;
+
+  return {
+    CustomComponent,
+    customProps,
+    customEditable,
+  };
+};
+
+const getElementPropsConfig = (
+  block: BlockNodeRecord,
+  editorKey: string,
+  offsetKey: string,
+  blockStyleFn: BlockStyleFn,
+  customConfig: *,
+): Object => {
+  let elementProps: Object = {
+    'data-block': true,
+    'data-editor': editorKey,
+    'data-offset-key': offsetKey,
+    key: block.getKey(),
+  };
+  const customClass = blockStyleFn(block);
+
+  if (customClass) {
+    elementProps.className = customClass;
+  }
+
+  if (customConfig.customEditable !== undefined) {
+    elementProps = {
+      ...elementProps,
+      contentEditable: customConfig.customEditable,
+      suppressContentEditableWarning: true,
+    };
+  }
+
+  return elementProps;
+};
+
+class DraftEditorBlockNode extends React.Component<Props> {
+  shouldComponentUpdate(nextProps: Props): boolean {
+    const {block, direction, tree} = this.props;
+
+    return (
+      block.getChildKeys() !== List() ||
+      block !== nextProps.block ||
+      tree !== nextProps.tree ||
+      direction !== nextProps.direction ||
+      (isBlockOnSelectionEdge(nextProps.selection, nextProps.block.getKey()) &&
```

Comment:
This line perhaps deserves a comment.  So why would we update if we're on a selection edge and want to forceSelection?

Reply:
This is legacy from the port of the other BlockNode